src/utils/ViewMapLeaf.ts

```diff
@@ -0,0 +1,96 @@
+export class ViewMapLeaf<T> {
+  public containerType?: ContainerType;
+  public parent: ViewMapLeaf<T>;
+  public childs: ViewMapLeaf<T>[];
+
+  private value?: T;
+
+  constructor () {
+    this.childs = [];
+  }
+
+  public isContainer (): boolean {
+    return this.getValue() === undefined;
+  }
```

Comment:
@shockone hey! I've tried today move `ViewMapLeaf` to `ViewMapLeaf` and `ViewMapBranch`and to be honest it looks super weird in the code. Typescript is not supporting now command `instanceof` for generic classes, so it is first weird place, another place is when we are finding recursively we need to have methods for each type and return only `ViewMapLeaf`.. so complexity is much bigger, readability much better.. so I decided to leave it now as it is.